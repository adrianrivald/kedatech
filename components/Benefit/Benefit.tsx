import Image from "next/image";
import React from "react";
import { twMerge } from "tailwind-merge";
import { maxWidthContainer } from "../../constants/class";
import { Fade } from "react-awesome-reveal";

export function Benefit() {
  return (
    <div id="benefit" className="mt-8 lg:mt-32">
      <div className="mt-48 flex flex-col lg:flex-row justify-between items-center gap-16 relative">
        <Fade direction="left" triggerOnce className=" w-full lg:w-[40%]">
          <div className="flex p-4 lg:ml-36 ml-0 flex-col gap-2 items-start">
            <h2
              role="benefit1heading"
              className="m-0 text-[30px] lg:text-[32px] lg:text-[40px] font-bold w-full"
            >
              Otomatisasi Pencatatan Barang
            </h2>
            <p
              role="benefit1desc"
              className="mt-4 lg:mt-0 m-0 text-[20px] w-full "
            >
              Tidak perlu lagi mencatat secara manual â€” semua barang masuk dan
              keluar tercatat otomatis & akurat.
            </p>
          </div>
        </Fade>
        <Fade direction="right" triggerOnce className=" w-full lg:w-[40%]">
          <div className="p-24 rounded-l-full bg-primary-100">
            <Image
              role="benefit1img"
              src="/assets/images/benefit-1.svg"
              width={612}
              height={408}
              alt="benefit-1"
              className="w-full lg:-ml-32"
            />
          </div>
        </Fade>
      </div>
      <div className="mt-48 flex flex-col lg:flex-row justify-between items-center gap-16 relative">
        <Fade direction="left" triggerOnce className=" w-full lg:w-[40%]">
          <div className="flex p-4 lg:ml-36 ml-0 flex-col gap-2 items-start">
            <h2
              role="benefit2heading"
              className="m-0 text-[30px] lg:text-[32px] lg:text-[40px] font-bold w-full"
            >
              Pantauan Keuntungan Harian
            </h2>
            <p
              role="benefit2desc"
              className="mt-4 lg:mt-0 m-0 text-[20px] w-full "
            >
              Lihat laporan keuntungan setiap hari secara real-time untuk
              memantau performa bisnis Anda dengan lebih baik.
            </p>
          </div>
        </Fade>
        <Fade direction="right" triggerOnce className=" w-full lg:w-[40%]">
          <div className="p-24 rounded-l-full bg-primary-100">
            <Image
              role="benefit2img"
              src="/assets/images/benefit-2.svg"
              width={612}
              height={408}
              alt="benefit-2"
              className="w-full lg:-ml-32"
            />
          </div>
        </Fade>
      </div>
      <div className="mt-48 flex flex-col lg:flex-row justify-between items-center gap-16 relative">
        <Fade direction="left" triggerOnce className=" w-full lg:w-[40%]">
          <div className="flex p-4 lg:ml-36 ml-0 flex-col gap-2 items-start">
            <h2
              role="benefit3heading"
              className="m-0 text-[30px] lg:text-[32px] lg:text-[40px] font-bold w-full"
            >
              Pengambilan Keputusan Lebih Cepat{" "}
            </h2>
            <p
              role="benefit3desc"
              className="mt-4 lg:mt-0 m-0 text-[20px] w-full "
            >
              Dengan data yang tersaji lengkap dan rapi, Anda dapat mengambil
              keputusan strategis tanpa ragu.
            </p>
          </div>
        </Fade>
        <Fade direction="right" triggerOnce className=" w-full lg:w-[40%]">
          <div className="p-24 rounded-l-full bg-primary-100">
            <Image
              role="benefit3img"
              src="/assets/images/benefit-3.svg"
              width={648}
              height={282}
              alt="benefit-3"
              className="w-full lg:-ml-32"
            />
          </div>
        </Fade>
      </div>
      {/* <Parallax pages={4}>
        <ParallaxLayer
          sticky={{ start: 1, end: 3 }}
          style={{ ...alignCenter, justifyContent: "flex-start" }}
        >
          <div className="ml-[15%]">
            <p>I'm a sticky layer</p>
          </div>
        </ParallaxLayer>
        <ParallaxLayer
          offset={1.5}
          speed={1.5}
          style={{ ...alignCenter, justifyContent: "flex-end" }}
        >
          <div className="mr-[15%]">
            <p>I'm not</p>
          </div>
        </ParallaxLayer>

        <ParallaxLayer
          offset={2.5}
          speed={1.5}
          style={{ ...alignCenter, justifyContent: "flex-end" }}
        >
          <div className="mr-[15%]">
            <p>Neither am I</p>
          </div>
        </ParallaxLayer>
      </Parallax> */}
    </div>
  );
}
