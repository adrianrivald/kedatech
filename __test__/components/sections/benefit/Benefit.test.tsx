import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import React from "react";
import renderer from "react-test-renderer";
import { Benefit } from "../../../../components/Benefit/Benefit";

it("Benefit snapshot test", () => {
  const tree = renderer.create(<Benefit />).toJSON();
  expect(tree).toMatchSnapshot();
});

beforeAll(() => {
  global.IntersectionObserver = class {
    constructor() {}
    observe() {}
    unobserve() {}
    disconnect() {}
  };
});

test("Benefit text inside test", () => {
  render(<Benefit />);

  // Benefit 1

  // Heading 1
  const benefit1headingEl = screen.getByRole("benefit1heading");
  expect(benefit1headingEl).toHaveTextContent("Otomatisasi Pencatatan Barang");
  // Desc 1
  const benefit1descEl = screen.getByRole("benefit1desc");
  expect(benefit1descEl).toHaveTextContent(
    "Tidak perlu lagi mencatat secara manual â€” semua barang masuk dan keluar tercatat otomatis & akurat."
  );
  // Image 1
  const benefit1imgEl = screen.getByRole("benefit1img");
  expect(benefit1imgEl).toHaveAttribute("src", "/assets/images/benefit-1.svg");

  // Benefit 2

  // Heading 2
  const benefit2headingEl = screen.getByRole("benefit2heading");
  expect(benefit2headingEl).toHaveTextContent("Pantauan Keuntungan Harian");
  // Desc 2
  const benefit2descEl = screen.getByRole("benefit2desc");
  expect(benefit2descEl).toHaveTextContent(
    "Lihat laporan keuntungan setiap hari secara real-time untuk memantau performa bisnis Anda dengan lebih baik."
  );
  // Image 2
  const benefit2imgEl = screen.getByRole("benefit2img");
  expect(benefit2imgEl).toHaveAttribute("src", "/assets/images/benefit-2.svg");

  // Benefit 3

  // Heading 3
  const benefit3headingEl = screen.getByRole("benefit3heading");
  expect(benefit3headingEl).toHaveTextContent(
    "Pengambilan Keputusan Lebih Cepat"
  );
  // Desc 3
  const benefit3descEl = screen.getByRole("benefit3desc");
  expect(benefit3descEl).toHaveTextContent(
    "Dengan data yang tersaji lengkap dan rapi, Anda dapat mengambil keputusan strategis tanpa ragu."
  );
  // Image 3
  const benefit3imgEl = screen.getByRole("benefit3img");
  expect(benefit3imgEl).toHaveAttribute("src", "/assets/images/benefit-3.svg");
});
