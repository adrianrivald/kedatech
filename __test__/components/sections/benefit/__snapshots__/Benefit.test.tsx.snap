// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Benefit snapshot test 1`] = `
<div
  className="mt-8 lg:mt-32"
  id="benefit"
>
  <div
    className="mt-48 flex flex-col lg:flex-row justify-between items-center gap-16 relative"
  >
    <div
      className="w-full lg:w-[40%] css-0"
      style={
        {
          "animationDelay": "0ms",
          "opacity": 0,
        }
      }
    >
      <div
        className="flex p-4 lg:ml-36 ml-0 flex-col gap-2 items-start"
        style={{}}
      >
        <h2
          className="m-0 text-[30px] lg:text-[32px] lg:text-[40px] font-bold w-full"
          role="benefit1heading"
        >
          Otomatisasi Pencatatan Barang
        </h2>
        <p
          className="mt-4 lg:mt-0 m-0 text-[20px] w-full "
          role="benefit1desc"
        >
          Tidak perlu lagi mencatat secara manual â€” semua barang masuk dan keluar tercatat otomatis & akurat.
        </p>
      </div>
    </div>
    <div
      className="w-full lg:w-[40%] css-0"
      style={
        {
          "animationDelay": "0ms",
          "opacity": 0,
        }
      }
    >
      <div
        className="p-24 rounded-l-full bg-primary-100"
        style={{}}
      >
        <img
          alt="benefit-1"
          className="w-full lg:-ml-32"
          data-nimg="1"
          decoding="async"
          height={408}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          role="benefit1img"
          src="/assets/images/benefit-1.svg"
          srcSet="/assets/images/benefit-1.svg 1x, /assets/images/benefit-1.svg 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={612}
        />
      </div>
    </div>
  </div>
  <div
    className="mt-48 flex flex-col lg:flex-row justify-between items-center gap-16 relative"
  >
    <div
      className="w-full lg:w-[40%] css-0"
      style={
        {
          "animationDelay": "0ms",
          "opacity": 0,
        }
      }
    >
      <div
        className="flex p-4 lg:ml-36 ml-0 flex-col gap-2 items-start"
        style={{}}
      >
        <h2
          className="m-0 text-[30px] lg:text-[32px] lg:text-[40px] font-bold w-full"
          role="benefit2heading"
        >
          Pantauan Keuntungan Harian
        </h2>
        <p
          className="mt-4 lg:mt-0 m-0 text-[20px] w-full "
          role="benefit2desc"
        >
          Lihat laporan keuntungan setiap hari secara real-time untuk memantau performa bisnis Anda dengan lebih baik.
        </p>
      </div>
    </div>
    <div
      className="w-full lg:w-[40%] css-0"
      style={
        {
          "animationDelay": "0ms",
          "opacity": 0,
        }
      }
    >
      <div
        className="p-24 rounded-l-full bg-primary-100"
        style={{}}
      >
        <img
          alt="benefit-2"
          className="w-full lg:-ml-32"
          data-nimg="1"
          decoding="async"
          height={408}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          role="benefit2img"
          src="/assets/images/benefit-2.svg"
          srcSet="/assets/images/benefit-2.svg 1x, /assets/images/benefit-2.svg 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={612}
        />
      </div>
    </div>
  </div>
  <div
    className="mt-48 flex flex-col lg:flex-row justify-between items-center gap-16 relative"
  >
    <div
      className="w-full lg:w-[40%] css-0"
      style={
        {
          "animationDelay": "0ms",
          "opacity": 0,
        }
      }
    >
      <div
        className="flex p-4 lg:ml-36 ml-0 flex-col gap-2 items-start"
        style={{}}
      >
        <h2
          className="m-0 text-[30px] lg:text-[32px] lg:text-[40px] font-bold w-full"
          role="benefit3heading"
        >
          Pengambilan Keputusan Lebih Cepat
           
        </h2>
        <p
          className="mt-4 lg:mt-0 m-0 text-[20px] w-full "
          role="benefit3desc"
        >
          Dengan data yang tersaji lengkap dan rapi, Anda dapat mengambil keputusan strategis tanpa ragu.
        </p>
      </div>
    </div>
    <div
      className="w-full lg:w-[40%] css-0"
      style={
        {
          "animationDelay": "0ms",
          "opacity": 0,
        }
      }
    >
      <div
        className="p-24 rounded-l-full bg-primary-100"
        style={{}}
      >
        <img
          alt="benefit-3"
          className="w-full lg:-ml-32"
          data-nimg="1"
          decoding="async"
          height={282}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          role="benefit3img"
          src="/assets/images/benefit-3.svg"
          srcSet="/assets/images/benefit-3.svg 1x, /assets/images/benefit-3.svg 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={648}
        />
      </div>
    </div>
  </div>
</div>
`;
